client/repoutils.c

```diff
@@ -259,46 +259,49 @@ TDNFRepoRemoveCache(
     BAIL_ON_TDNF_ERROR(dwError);
 
     pDir = opendir(pszRepoCacheDir);
-    if(pDir == NULL)
+    if(pDir == NULL && errno != ENOENT)
     {
         dwError = errno;
         BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
     }
 
-    while ((pEnt = readdir (pDir)) != NULL )
+    if (pDir != NULL)
```

Comment:
This works. Please check https://man7.org/linux/man-pages/man3/nftw.3p.html
Please remove the code duplication TDNFRemoveRpmCache & TDNFRepoRemoveCache functions. We can have a single function to delete a directory.

Reply:
I think I agree, we should have a single function to recursively remove directories. I have just found that RPMs will not be removed at all unless they are in a flat `x86_64` or `noarch` directory, see https://github.com/vmware/tdnf/issues/213. If the structure is different, RPMs will not be cleaned. Fixing this is more complex, and should go into a separate change.

Reply:
Directories are now removed recursively.